com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs

```diff
@@ -284,12 +286,27 @@ internal void NetworkSpawnObject(SnapshotSpawnCommand spawnCommand, ulong srcCli
                     spawnCommand.ObjectRotation);
             }
 
-            m_NetworkManager.SpawnManager.SpawnNetworkObjectLocally(networkObject, spawnCommand.NetworkObjectId,
+            m_NetworkManager.SpawnManager.PreSpawnNetworkObjectLocallyCommon(networkObject, spawnCommand.NetworkObjectId,
                 true, spawnCommand.IsPlayerObject, spawnCommand.OwnerClientId, false);
+
             //todo: discuss with tools how to report shared bytes
             m_NetworkManager.NetworkMetrics.TrackObjectSpawnReceived(srcClientId, networkObject, 8);
         }
 
+        internal void NetworkPostSpawnObject(SnapshotSpawnCommand spawnCommand, ulong srcClientId)
```

Comment:
let's have more comments on both this and the other counterpart one above?

Reply:
Added comments.